### PR TITLE
Add OTLP trace with ADC example

### PR DESCRIPTION
**Pull request by damemi**
Cobbled this together following docs from:

* https://opentelemetry.io/docs/languages/java/exporters/
* https://github.com/googleapis/google-auth-library-java
* https://github.com/googleapis/google-auth-library-java/blob/main/samples/snippets/src/main/java/AuthenticateExplicit.java
* https://cloud.google.com/java/docs/reference/google-auth-library/latest/com.google.auth.oauth2.GoogleCredentials
* https://javadoc.io/static/io.opentelemetry/opentelemetry-exporter-otlp-http-trace/1.14.0/io/opentelemetry/exporter/otlp/http/trace/OtlpHttpSpanExporterBuilder.html

Same idea as https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/pull/800, https://github.com/GoogleCloudPlatform/opentelemetry-operations-python/pull/307, and https://github.com/GoogleCloudPlatform/opentelemetry-operations-js/pull/676

### CODE REVIEW

**Comment by psx95** [comment]
Is this different from [this example](https://github.com/GoogleCloudPlatform/opentelemetry-operations-java/blob/main/examples/trace/src/main/java/com/google/cloud/opentelemetry/example/trace/TraceExporterExample.java) which is already present ?

It seems similar.

**Comment by psx95** [comment]
> Is this different from [this example](https://github.com/GoogleCloudPlatform/opentelemetry-operations-java/blob/main/examples/trace/src/main/java/com/google/cloud/opentelemetry/example/trace/TraceExporterExample.java) which is already present ?
> 
> It seems similar.

Ah, nvm it uses OTLP Span exporter instead of Google Cloud Exporter.

**Review by psx95 — COMMENTED** [review]
The sample looks good. A couple of follow-ups: 
 - Should we move it to the existing `examples` folder ? The proposed example is placed in a folder called `example`. 
  - Could we add a corresponding gradle file along with a small Readme with instructions to build and run this example so that it remains consistent with the rest of the examples.

**Review by psx95 — CHANGES_REQUESTED** [review]
Looks good to me! Thanks @damemi ! 👍🏻 

Following from our offline discussion, two pending action items prior to merging - 
 - Adding `-Dotel.exporter.otlp.protocol=http/protobuf'` to the `autoconf_config` in `build.gradle`
 -  Adding a README file under `examples/otlptrace` to give an overview of this example and instructions on running it.

**Comment by damemi** [comment]
Thanks for all your help @psx95! I added the changes we talked about and tested this and it's working now 🎉

File: examples/otlptrace/src/main/java/com/google/cloud/opentelemetry/example/otlptrace/OTLPTraceExample.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        credentials.refreshIfExpired();
+        OtlpHttpSpanExporterBuilder builder =
+            ((OtlpHttpSpanExporter) exporter)
+                .toBuilder()
+                    .addHeader(
+                        "Authorization", "Bearer " + credentials.getAccessToken().getTokenValue());
+
+        return builder.build();
+      } catch (IOException e) {
+        System.out.println("error");
```
Comment (psx95) [inline examples/otlptrace/src/main/java/com/google/cloud/opentelemetry/example/otlptrace/OTLPTraceExample.java:64]: Nit: Could be better to log the underlying error message as well 

```
System.out.println("error:  " + e.getMessage());
```

**Review by psx95 — APPROVED** [review]

**Pull request merged**